Mango: creation and listing of indexes

robertkowalski opened the pull request:
Part 1/2 for Mango:

Creation of Mango indexes and listing them. Disabled pagination
and bulk-deletion for now, see:

https://issues.apache.org/jira/browse/COUCHDB-2651
https://issues.apache.org/jira/browse/COUCHDB-2652

As you can't use the feature right now (querying is part 2/2)
the buttons to reach the features are hidden behind a feature flip.

Use the direct urls to access the features:

http://localhost:8000/#database/$YOUR_DATABASE/_index
http://localhost:8000/#database/$YOUR_DATABASE/_indexlist

or add `?beta=showquery` to your url, e.g.

http://localhost:8000/#database/$YOUR_DATABASE/_all_docs?beta=showquery

Additionally prepares the app for i18n.

COUCHDB-2627

garrensmith commented on app/addons/components/assets/less/header-togglebutton.less:
```diff
@@ -11,6 +11,8 @@
 // the License.
 @import "../../../../../assets/less/variables.less";
 
+@import "../../../../../assets/less/variables.less";
```
You do need to import this twice?

garrensmith commented on app/addons/components/tests/docSpec.react.jsx:
```diff
@@ -81,12 +81,23 @@ define([
       assert.ok(spy.calledOnce);
     });
+
+    it('it can render without checkbox', function () {
```
You don't need the extra 'it' in the statement. Its there already :smile:

garrensmith commented on app/addons/documents/base.js:
```diff
@@ -110,5 +110,24 @@ function (app, FauxtonAPI, Documents) {
+
+      return 'database/' + db + '/_index' + query;
+    }
+  });
```
Don't you need a url for the server?

robertkowalski commented:
@garrensmith  i removed the flip and added a help screen instead of a list of indexes:

![bildschirmfoto 2015-03-31 um 16 06 54](https://cloud.githubusercontent.com/assets/298166/6920650/00fb76b4-d7c0-11e4-9ad2-4627ace1f70a.png)

garrensmith commented:
+1 once tests pass

robertkowalski commented:
merged as d7641f16e7a12faff896136826b43f7affb0a393
